x/rollup/keeper/msg_server_test.go

```diff
@@ -0,0 +1,236 @@
+	contractCreationTxBz, err := contractCreationTx.MarshalBinary()
+	s.Require().NoError(err)
+	invalidTxBz := []byte("invalid tx bytes")
+
+	tests := map[string]struct {
+		msg         *types.MsgApplyL1Txs
+		setupMocks  func()
+		shouldError bool
+	}{
+		"successful message with single user deposit tx": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, depositTxBz},
+			},
+			shouldError: false,
+		},
+		"successful message with multiple user deposit txs": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, depositTxBz, depositTxBz},
+			},
+			shouldError: false,
+		},
+		"invalid l1 attributes tx bytes": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{invalidTxBz, depositTxBz},
+			},
+			shouldError: true,
+		},
+		"non-deposit tx passed in as l1 attributes tx": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{cosmosEthTxBz, depositTxBz},
+			},
+			shouldError: true,
+		},
+		"user deposit tx passed in as l1 attributes tx": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{depositTxBz, depositTxBz},
+			},
+			shouldError: true,
+		},
+		"invalid user deposit tx bytes": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, invalidTxBz},
+			},
+			shouldError: true,
+		},
+		"non-deposit tx passed in as user deposit tx": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, cosmosEthTxBz},
+			},
+			shouldError: true,
+		},
+		"l1 attributes tx passed in as user deposit tx": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, l1AttributesTxBz},
+			},
+			shouldError: true,
+		},
+		"contract creation tx passed in as user deposit tx": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, contractCreationTxBz},
+			},
+			shouldError: true,
+		},
+		"one successful l1 user deposit tx and an invalid tx passed in as user deposit txs": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, depositTxBz, invalidTxBz},
+			},
+			shouldError: true,
+		},
+		"bank keeper mint coins failure": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, depositTxBz},
+			},
+			setupMocks: func() {
+				s.bankKeeper.EXPECT().MintCoins(gomock.Any(), types.ModuleName, gomock.Any()).Return(sdkerrors.ErrUnauthorized)
+			},
+			shouldError: true,
+		},
+		"bank keeper send coins failure": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, depositTxBz},
+			},
+			setupMocks: func() {
+				s.bankKeeper.EXPECT().SendCoinsFromModuleToAccount(gomock.Any(), types.ModuleName, gomock.Any(), gomock.Any()).Return(sdkerrors.ErrUnknownRequest)
+			},
+			shouldError: true,
+		},
+	}
+
+	for name, test := range tests {
+		s.Run(name, func() {
+			if test.setupMocks != nil {
+				test.setupMocks()
+			}
+			s.mockMintETH()
+
+			var resp *types.MsgApplyL1TxsResponse
+			resp, err = keeper.NewMsgServerImpl(s.rollupKeeper).ApplyL1Txs(s.ctx, test.msg)
```

Comment:
Why the extra `var` and not `:=`?

Reply:
I was getting the following IDE warning and figured I'd remove the shadowed declaration and just reassign the previously initialized err field: `Declaration of 'err' shadows declaration at msg_server_test.go`

It's kind of unfortunate you can't conditionally initialize vars in go. Lmk if you feel strongly about changing this, otherwise I'll leave it in

Reply:
It'll be removed in the integration test [PR](https://github.com/polymerdao/monomer/pull/165/files#diff-4153058257cbc126a783b5411a175afa9a7b2a5116bd1685cae4d62526f43113R97) too since now `err` isn't declared above it :)